From b71031076e1169e89bdac1b245ad1488587e4730 Mon Sep 17 00:00:00 2001
From: Maciej Trela <Maciej.Trela@intel.com>
Date: Wed, 14 Apr 2010 16:58:16 +1000
Subject: [PATCH] md: Correctly handle device removal via sysfs

Writing "none" to "../md/dev-xx/slot" removes that device
from being an active part of the array, but it didn't
set ->raid_disk to -1 to record this fact.


Signed-off-by: Maciej Trela <Maciej.Trela@intel.com>
Signed-off-by: NeilBrown <neilb@suse.de>
---
 drivers/md/md.c |    1 +
 1 file changed, 1 insertion(+)

--- linux-2.6.32-SLE11-SP1.orig/drivers/md/md.c
+++ linux-2.6.32-SLE11-SP1/drivers/md/md.c
@@ -2402,6 +2402,7 @@ slot_store(mdk_rdev_t *rdev, const char
 			return err;
 		sprintf(nm, "rd%d", rdev->raid_disk);
 		sysfs_remove_link(&rdev->mddev->kobj, nm);
+		rdev->raid_disk = -1;
 		set_bit(MD_RECOVERY_NEEDED, &rdev->mddev->recovery);
 		md_wakeup_thread(rdev->mddev->thread);
 	} else if (rdev->mddev->pers) {
